src/java.desktop/share/native/libfontmanager/freetypeScaler.c

```diff
@@ -1285,7 +1287,7 @@ static int allocateSpaceForGP(GPData* gpdata, int npoints, int ncontours) {
     }
 
     /* failure if any of mallocs failed */
-    if (gpdata->pointTypes == NULL ||  gpdata->pointCoords == NULL)
+    if (gpdata->pointTypes == NULL || gpdata->pointCoords == NULL)
```

Comment:
I think it would be cleaner to free the remaining allocations on the failing path:

```
if (gpdata->pointTypes == NULL || gpdata->pointCoords == NULL) {
   if (gpdata->pointTypes != NULL)  free(gpdata->pointTypes);
   if (gpdata->pointCoords != NULL) free(gpdata->pointCoords);
   return 0;
} else {
   return 1;
}
```

Reply:
then we would free   as well  for the realloc code path
    } else {
        /* do we have enough space? */
    ...
    }
Is this okay?

Reply:
Yes, I think that is better: `realloc` can fail as well, and we don't want to leak those?

Reply:
Please also reset the free'd pointers to NULL in GPData* before returning.

Reply:
Hello  Thomas and Aleksey, I moved the free calls and added the NULL settings. Are you fine with the latest commit?
Thanks, Matthias